source/ingress.go

```diff
@@ -226,6 +231,24 @@ func (sc *ingressSource) setDualstackLabel(ingress *networkv1.Ingress, endpoints
 	}
 }
 
+func endpointsForIngressHostname(hostname string, targets endpoint.Targets, ttl endpoint.TTL, providerSpecific endpoint.ProviderSpecific, setIdentifier string) []*endpoint.Endpoint {
+	var endpoints []*endpoint.Endpoint
+	if len(targets) == 0 {
+		epNull := &endpoint.Endpoint{
+			DNSName:          strings.TrimSuffix(hostname, "."),
+			Targets:          targets,
+			RecordTTL:        ttl,
+			RecordType:       endpoint.RecordTypeA,
```

Comment:
what if it's a CNAME record?

Reply:
In this case, we are only providing an endpoint with no targets so it is skipped in the plan.
We provided a record type just so there's any data in the field to avoid `nil` surprises.